sdk/dotnet/Pulumi/Core/Output.cs

```diff
@@ -73,7 +76,10 @@ public static Output<ImmutableArray<T>> All<T>(params Output<T>[] outputs)
         /// <see cref="All{T}(Output{T}[])"/> for more details.
         /// </summary>
         public static Output<ImmutableArray<T>> All<T>(IEnumerable<Output<T>> outputs)
-            => Output<T>.All(ImmutableArray.CreateRange(outputs.Select(o => (Input<T>)o)));
+            => Output<T>.All(ImmutableArray.CreateRange((outputs switch {
```

Comment:
Hmm, this is interesting.  

Something here strikes me as very unfortunate. The function expects IEnumerable and it so happens that one of the many subtypes of that, ImmutableArray admits a default value that breaks the IEnumerable contracts, so the PR suggest to detect and patch that. However, the problem is that there's a potentially infinite variety of invalid IEnumerable implementations that break the contract, and it would not be possible to detect them all here (the switch will get too large). What I am saying perhaps the fault belongs to the caller of this function and should be fixed there, not in this function?

Do we know where the default ImmutableArray originated, where was it constructed?

If we were to edit this function, perhaps a null check plus meaningful failure would do? Still I feel it might be a bit too defensive given that IEnumerable is a very common thing to use and typically there are no null checks around it.

Reply:
After thinking about it more carefully, I think the changes to `Output.All` should be reverted, as indeed it is the caller of the function who is responsible for providing a valid `IEnumerable`.